### PR TITLE
Fi-related hmac_drbg fixes

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This PR introduces changes related to FI in hmac_drbg. It also fixes a bug introduced in https://github.com/ARMmbed/mbedtls/pull/2934.

### CODE REVIEW

**Comment by AndrzejKurek** [comment]
Updated with an additional commit regarding `no_reseed` environments.

File: library/hmac_drbg.c [hunk 1]
```diff
@@ -62,6 +62,8 @@ void mbedtls_hmac_drbg_init( mbedtls_hmac_drbg_context *ctx )
 {
     memset( ctx, 0, sizeof( mbedtls_hmac_drbg_context ) );
 
+    ctx->prediction_resistance = MBEDTLS_HMAC_DRBG_PR_OFF;
+    ctx->reseed_flag = MBEDTLS_HMAC_DRBG_RESEED;
```
Comment (piotr-now) [inline library/hmac_drbg.c:66]: I wonder if setting the hamming distance for a value that is never checked makes sense. The `ctx->reseed_flag` could have any value other than MBEDTLS_HMAC_DRBG_NO_RESEED and its behavior will be always the same.

File: library/hmac_drbg.c [hunk 2]
```diff
@@ -387,12 +389,25 @@ void mbedtls_hmac_drbg_set_prediction_resistance( mbedtls_hmac_drbg_context *ctx
     ctx->prediction_resistance = resistance;
 }
 
+/*
+ * Set the reseeding flag
+ */
+void mbedtls_hmac_drbg_set_reseeding( mbedtls_hmac_drbg_context *ctx,
+                                      int reseed_flag )
+{
+    ctx->reseed_flag = reseed_flag;
```
Comment (piotr-now) [inline library/hmac_drbg.c:398]: If this function takes only two values, it is worth adding an assert. Or change the meaning of it into for example:
`void mbedtls_hmac_drbg_set_reseeding( mbedtls_hmac_drbg_context *ctx, int enable )`. Then for better encapsulation you could move `MBEDTLS_HMAC_DRBG_NO_RESEED` to * .c file and set it inside this function depending on enable

File: tests/suites/test_suite_hmac_drbg.function [hunk 3]
```diff
@@ -219,7 +220,8 @@ void hmac_drbg_nopr( int md_alg, data_t * entropy, data_t * custom,
     mbedtls_hmac_drbg_context ctx;
 
     mbedtls_hmac_drbg_init( &ctx );
-
+	mbedtls_hmac_drbg_set_reseeding( &ctx, MBEDTLS_HMAC_DRBG_NO_RESEED );
+
```
Comment (piotr-now) [inline tests/suites/test_suite_hmac_drbg.function:224]: Tab instead of spaces

File: tests/suites/test_suite_hmac_drbg.function [hunk 4]
```diff
@@ -172,7 +173,7 @@ void hmac_drbg_no_reseed( int md_alg, data_t * entropy,
     mbedtls_hmac_drbg_context ctx;
 
     mbedtls_hmac_drbg_init( &ctx );
-
+	mbedtls_hmac_drbg_set_reseeding( &ctx, MBEDTLS_HMAC_DRBG_NO_RESEED );
```
Comment (piotr-now) [inline tests/suites/test_suite_hmac_drbg.function:176]: Tab instead of spaces

File: library/hmac_drbg.c (hunk 1 above)
Comment (AndrzejKurek) [reply library/hmac_drbg.c:66]: The initial value is `MBEDTLS_HMAC_DRBG_RESEED`, and the check is for `!= MBEDTLS_HMAC_DRBG_NO_RESEED` so if a bad guy wants to skip reseeding, he has to change a lot of bits.

File: tests/suites/test_suite_hmac_drbg.function (hunk 4 above)
Comment (AndrzejKurek) [reply tests/suites/test_suite_hmac_drbg.function]: Fixed.

File: tests/suites/test_suite_hmac_drbg.function (hunk 3 above)
Comment (AndrzejKurek) [reply tests/suites/test_suite_hmac_drbg.function]: Fixed.

**Comment by AndrzejKurek** [comment]
Removed four spaces from a .function file to make the CI happy.

**Comment by AndrzejKurek** [comment]
Travis failures are a known issue, while the merge job fails on API/ABI change.

**Pull request merged**